plugins/by-name/nvim-ufo/default.nix

```diff
@@ -1,87 +1,133 @@
 {
   lib,
-  helpers,
-  config,
-  pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.nvim-ufo;
+  inherit (lib.nixvim) defaultNullOpts;
 in
-{
-  options.plugins.nvim-ufo = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "nvim-ufo";
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "nvim-ufo";
+  luaName = "ufo";
+  package = "nvim-ufo";
 
-    package = lib.mkPackageOption pkgs "nvim-ufo" {
-      default = [
-        "vimPlugins"
-        "nvim-ufo"
-      ];
-    };
+  maintainers = [ lib.maintainers.khaneliman ];
 
-    openFoldHlTimeout = helpers.defaultNullOpts.mkInt 400 ''
+  # TODO: added 2024-10-05 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "openFoldHlTimeout"
+    "providerSelector"
+    "closeFoldKinds"
```

Comment:
Isn't the new name `close_fold_kinds_for_ft` ? If so, you need to manually add an `mkRenamedOptionSubmodule` to `imports`.

Reply:
It is deprecated and not removed. They are different types, as well. This is a rename for compatibility but the documented option is the new name.

Reply:
https://github.com/kevinhwang91/nvim-ufo/commit/7b14dd650b57291a49f037f1ede325d46e08b258

Reply:
Ok I see. So we do want the values given to `closeFoldKinds` to be forwarded to `settings.close_fold_kinds` then, and let the plugin warn the user about the deprecation. I have correctly understood ?

If so, your implem looks good.